pylint/lint.py

```diff
@@ -1054,31 +1082,57 @@ def _do_check(self, files_or_modules):
             if interfaces.implements(checker, interfaces.IAstroidChecker):
                 walker.add_checker(checker)
         # build ast and check modules or packages
-        for descr in self.expand_files(files_or_modules):
-            modname, filepath, is_arg = descr["name"], descr["path"], descr["isarg"]
-            if not self.should_analyze_file(modname, filepath, is_argument=is_arg):
-                continue
+        if self.config.from_stdin:
+            assert len(files_or_modules) == 1
+            filepath = files_or_modules[0]
+            try:
+                # Note that modpath_from_file does not really perform an
+                # __import__ as one may expect, because ImportError
```

Comment:
The comment seems a bit confusing, indeed the function doesn't perform an `__import__`, but it raises an `ImportError`, which is why we should handle it here. Maybe rephrase the comment a bit?